### ISSUE TITLE
Support for Westmint / Assets

### ISSUE THREAD

**Issue by chevdor** [paritytech/substrate-matrix-faucet#151]
Westmint allows minting custom assets.

It would faciliate users' entry to be able to get minted assets directly on Westmint.
This will however first require the ability to drip Westmint WND so the recipient account has a sufficient balance to then receive minted Assets.

The goal is not to support a huge number of minted assets but we should plan on supporting a handful of them.

**Comment by mutantcornholio** [comment]
Hi! @chevdor 

Am I understanding it correctly that all we need is to teleport westies to Asset Hub parachain? 
For that, we already have it in https://paritytech.github.io/polkadot-testnet-faucet/westend.  
Is it enough?

**Comment by chevdor** [comment]
Teleporting is an option but not the only one and we could make it simpler by having a faucet account on the Asset Hub already funded with WND (from a manual or automatic teleport).

The goal is:
- if the target account has a sufficient balance, do nothing
- if the target account has unsufficient balance, drip a small amount to make the account a valid receiver for random assets
- drip the requested asset to the user's account

**Comment by mutantcornholio** [comment]
> drip the requested asset to the user's account

This one's a bit puzzling to me.  
If somebody had created an asset on Westmint, they own the supply, right? How can we drip it without having access to it?

**Comment by chevdor** [comment]
The goal is not to drip user's tokens... but one/some of our test tokens so people get a feel for assets.

**Comment by chevdor** [comment]
yes, but it misses the step where the user actually gets the custom asset, like one of those:
<img width="1407" alt="image" src="https://github.com/paritytech/polkadot-testnet-faucet/assets/738724/3d258541-0eed-4ac6-a0f6-da1e010ded8a">

As you pointed out, it cannot be a random one and needs to be one the faucet has "in stock" (or mint itself).

**Comment by mutantcornholio** [comment]
Waait, I think I see it now... Is it a usecase like this?
* you're making faucet the issuer of CHVTKN (or sending some CHVTKN to faucet) as an api call, and others can get it directly from the faucet?

**Comment by chevdor** [comment]
Yeah something like that. It could be `CHVTKN` or any... I could send some of those to the faucet and let it drip for users or we let the faucet manage an asset. By `any` I don't mean randomly any token, this is something we fix and once fixed, the list of drippable assets is fixed.

**Comment by mutantcornholio** [comment]
> By any I don't mean randomly any token, this is something we fix

Why not any? I'd say it would be more useful to let users distribute their own custom assets via faucet, rather than make a list of test assets that we choose. 

It will take some refactoring to get it, but totally solvable together with this issue: https://github.com/paritytech/polkadot-testnet-faucet/issues/278

**Comment by chevdor** [comment]
> Why not any? 

I was about to answer that we don't control them but indeed, as long as we have some positive balance, we could handle any and even let users send their assets to the faucet to add them to the list. That would be a neat idea. That works only for fungible tokens.

Let me know if some of the assets I minted can help.
<img width="446" alt="image" src="https://github.com/paritytech/polkadot-testnet-faucet/assets/738724/d407d05f-df96-4820-a392-fb33982fd035">

I will not mint more of the asset \#2 for obvious reasons but you can have some and I can mint more of the aset \#1.

We could extend to NFT but this was not the scope of this issue so I suggest to focus on assets for now.
